File: C++/008.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main(){
  int adjacent = 13;
  int length = 1000;
  long long max = 0;
  char nums[length];
  ifstream file;
  file.open("storage/8.txt");
  file >> nums;
  for (int i=0; i<length-adjacent; i++){
    long long int prod = 1;
    for (int j=0; j<adjacent; j++){
      prod *= (nums[i+j]-48);
    }
    if (prod > max){
      max = prod;
    }
  }
  cout << max;
}

File: Python/008.py
from Utils import product

with open("storage//8.txt") as f:
    number = f.read().replace("\n","")

max = 0
adjacent_digits = 13
for i in range(len(number)-adjacent_digits):
    prod = product(int(number[j]) for j in range(i, i+adjacent_digits))
    if prod > max:
        max = prod
print(max)
